google-http-client/src/test/java/com/google/api/client/http/MultipartContentTest.java

```diff
@@ -26,55 +26,76 @@
+    assertTrue(boundaryString.startsWith(BOUNDARY));
+    assertTrue(boundaryString.endsWith("__"));
+    assertEquals("multipart/related; boundary=" + boundaryString, content.getType());
+
+    final String[][] VALUES = new String[][] {
+            {"Hello world", "text/plain"},
+            {"<xml>Hi</xml>", "application/xml"},
+            {"{x:1,y:2}", "application/json"}
+    };
+    StringBuilder expectedContentSB = new StringBuilder();
+    for (String[] valueTypePair: VALUES) {
+      String contentValue = valueTypePair[0];
+      String contentType = valueTypePair[1];
+      content.addPart(new MultipartContent.Part(ByteArrayContent.fromString(contentType, contentValue)));
+      expectedContentSB.append("--").append(boundaryString).append(CRLF)
+              .append(headers(contentType, contentValue)).append(CRLF)
+              .append(contentValue).append(CRLF);
+    }
+    expectedContentSB.append("--").append(boundaryString).append("--").append(CRLF);
+    // write to string
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    content.writeTo(out);
+    String expectedContent = expectedContentSB.toString();
+    assertEquals(expectedContent, out.toString());
+    assertEquals(StringUtils.getBytesUtf8(expectedContent).length, content.getLength());
+  }
 
   public void testContent() throws Exception {
-    subtestContent("--__END_OF_PART__--" + CRLF, null);
+    subtestContent("--" + BOUNDARY + "--" + CRLF, null);
     subtestContent(
-        "--__END_OF_PART__" + CRLF + HEADERS + CRLF + "foo" + CRLF + "--__END_OF_PART__--" + CRLF,
-        null,
+        "--" + BOUNDARY + CRLF
+                + HEADERS + CRLF
+                + "foo" + CRLF
+                + "--" + BOUNDARY + "--" + CRLF,
+            null,
         "foo");
     subtestContent(
-        "--__END_OF_PART__"
-            + CRLF
-            + HEADERS
-            + CRLF
-            + "foo"
-            + CRLF
-            + "--__END_OF_PART__"
-            + CRLF
-            + HEADERS
-            + CRLF
-            + "bar"
-            + CRLF
-            + "--__END_OF_PART__--"
-            + CRLF,
-        null,
+        "--" + BOUNDARY + CRLF
+            + HEADERS + CRLF
+            + "foo" + CRLF
+            + "--" + BOUNDARY + CRLF
+            + HEADERS + CRLF
+            + "bar" + CRLF
+            + "--" + BOUNDARY + "--" + CRLF,
+            null,
         "foo",
         "bar");
     subtestContent(
-        "--myboundary"
-            + CRLF
-            + HEADERS
-            + CRLF
-            + "foo"
-            + CRLF
-            + "--myboundary"
-            + CRLF
-            + HEADERS
-            + CRLF
-            + "bar"
-            + CRLF
-            + "--myboundary--"
-            + CRLF,
+        "--myboundary" + CRLF
+            + HEADERS + CRLF
+            + "foo" + CRLF
+            + "--myboundary" + CRLF
+            + HEADERS + CRLF
+            + "bar" + CRLF
+            + "--myboundary--" + CRLF,
```

Comment:
After running that build I believe these files get updated automatically, If there is any modification after build then please add them as well.